gateway-shell/src/main/java/org/apache/knox/gateway/shell/KnoxSh.java

```diff
@@ -155,6 +171,84 @@ public boolean validate() {
     public abstract String getUsage();
   }
 
+  private class KnoxBuildTrustStore extends Command {
+
+    private static final String USAGE = "buildTrustStore --gateway server-url";
+    private static final String DESC = "Downloads the gateway server's public certificate and builds a trust store.";
+    private static final String GATEWAY_CERT_NOT_EXPORTED = "Finished work without building truststore";
+    private static final String GATEWAY_CERT_EXPORTED_MESSAGE_PREFIX = "Gateway server's certificate is exported into ";
+
+    @Override
+    public void execute() throws Exception {
+      String result = GATEWAY_CERT_NOT_EXPORTED;
+      try {
+        final X509Certificate gatewayServerPublicCert = fetchPublicCertFromGatewayServer();
+        if (gatewayServerPublicCert != null) {
+          final File trustStoreFile = Paths.get(System.getProperty("user.home"), KnoxSession.GATEWAY_CLIENT_TRUST).toFile();
+          X509CertificateUtil.writeCertificateToJks(gatewayServerPublicCert, trustStoreFile);
+          result = GATEWAY_CERT_EXPORTED_MESSAGE_PREFIX + trustStoreFile.getAbsolutePath();
+        }
+      } catch(Exception e) {
+        //NOP
```

Comment:
I think in this case, we aren't printing anything on failure. Previously the else part of the if/else would not have been reached since the exception would be thrown by `execute()`. Do we need the try/catch? Can we let the exception propagate? Maybe we should wrap the exception with a nicer message?

Reply:
Agreed; will submit the new patch soon